packages/bs-solana/src/services/blockchain-data/TatumRpcBDSSolana.ts

```diff
@@ -0,0 +1,436 @@
+import {
+  BalanceResponse,
+  BlockchainDataService,
+  ContractResponse,
+  formatTokenNumber,
+  Network,
+  RpcResponse,
+  Token,
+  TransactionResponse,
+  TransactionsByAddressParams,
+  TransactionsByAddressResponse,
+  TransactionTransferAsset,
+  TransactionTransferNft,
+} from '@cityofzion/blockchain-service'
+import { BSSolanaConstants, BSSolanaNetworkId } from '../../constants/BSSolanaConstants'
+import solanaSDK from '@solana/web3.js'
+import * as solanaSplSDK from '@solana/spl-token'
+import { BSSolanaCachedMethodsHelper } from '../../helpers/BSSolanaCachedMethodsHelper'
+import { BSSolanaHelper } from '../../helpers/BSSolanaHelper'
+
+export class TatumRpcBDSSolana implements BlockchainDataService {
+  static tatumUrlByNetworkId: Record<BSSolanaNetworkId, string> = {
+    'mainnet-beta': 'https://solana-mainnet.gateway.tatum.io/',
+    devnet: 'https://solana-devnet.gateway.tatum.io',
+  }
+
+  static getTatumConnection(network: Network<BSSolanaNetworkId>, apiKey: string) {
+    return new solanaSDK.Connection(this.tatumUrlByNetworkId[network.id], {
+      httpHeaders: {
+        'x-api-key': apiKey,
+      },
+    })
+  }
+
+  maxTimeToConfirmTransactionInMs: number = 1000 * 60
+
+  #network: Network<BSSolanaNetworkId>
+  #feeToken: Token
+
+  #connection: solanaSDK.Connection
+  #mainnetApiKey: string
+  #testnetApiKey: string
+
+  #functionByProgramIdAndMethod: Record<
+    string,
+    Record<
+      string,
+      (
+        instruction: solanaSDK.ParsedInstruction,
+        allInstructions: solanaSDK.ParsedInstruction[]
+      ) => Promise<TransactionTransferAsset | TransactionTransferNft>
+    >
+  > = {
+    [solanaSplSDK.TOKEN_PROGRAM_ID.toString()]: {
+      transferChecked: this.#parseSplTransferCheckedInstruction.bind(this),
+      transfer: this.#parseSplTransferInstruction.bind(this),
+    },
+    [solanaSDK.SystemProgram.programId.toString()]: {
+      transfer: this.#parseSystemInstruction.bind(this),
+    },
+  }
+
+  constructor(network: Network<BSSolanaNetworkId>, feeToken: Token, mainnetApiKey: string, testnetApiKey: string) {
```

Comment:
What do you think receive an object here?

Reply:
I'm following the pattern of other services. I agree that it's better to use object, but I think we could create another issue to change everywhere